runtime/root.go

```diff
@@ -0,0 +1,97 @@
+/*
+Copyright 2021 cuisongliu@qq.com.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package runtime
+
+import (
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	"github.com/alibaba/sealer/common"
+	"github.com/alibaba/sealer/logger"
+	v1 "github.com/alibaba/sealer/types/api/v1"
+	"github.com/alibaba/sealer/utils"
+)
+
+func (d *Default) CopyFilesOnRoot(cluster *v1.Cluster) error {
+	d.addMaster0Host(cluster)
+	d.copyKubeConfig(cluster)
+	d.copyKubectl(cluster)
+	return nil
+}
+
+func (d *Default) addMaster0Host(cluster *v1.Cluster) {
+	content, err := utils.ReadAll(common.EtcHosts)
+	if err != nil {
+		return
+	}
+	if !strings.Contains(string(content), common.APIServerDomain) {
+		err = utils.AppendFile(common.EtcHosts, fmt.Sprintf("%s %s", cluster.Spec.Masters.IPList[0], common.APIServerDomain))
+		if err != nil {
+			logger.Warn("append master0 host to etc hosts failed: %v", err)
+		}
+	}
+}
+func (d *Default) copyKubeConfig(cluster *v1.Cluster) {
+	if err := utils.MkDirIfNotExists(common.DefaultKubeConfigDir()); err != nil {
+		logger.Warn("mkdir kube dir failed: %v", err)
+	}
+	adminConf := filepath.Join(common.DefaultClusterRootfsDir, cluster.Name, "admin.conf")
+	if !utils.IsFileExist(adminConf) {
+		adminConf = common.KubeAdminConf
+	}
+	kubeConfig := filepath.Join(common.DefaultKubeConfigDir(), "config")
+	if !utils.IsFileExist(kubeConfig) {
+		_, err := utils.CopySingleFile(adminConf, kubeConfig)
+		if err != nil {
+			logger.Warn("copy kube config failed: %v", err)
+		}
+	}
+}
+func (d *Default) copyKubectl(cluster *v1.Cluster) {
```

Comment:
I found that in this function implementation, you only fetch Name in cluster, the why not only define it as `func (d *Default) copyKubectl(clusterName string)`

It would be much more cohesive if you grant the access right when it only needs.